x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/filtering/FilteringAdvancedSnippet.java

```diff
@@ -84,7 +80,14 @@ public FilteringAdvancedSnippet(StreamInput in) throws IOException {
             UPDATED_AT_FIELD,
             ObjectParser.ValueType.STRING
         );
-        PARSER.declareField(constructorArg(), (p, c) -> p.map(), VALUE_FIELD, ObjectParser.ValueType.OBJECT);
+        PARSER.declareField(constructorArg(), (p, c) -> {
+            if (p.currentToken() == XContentParser.Token.START_ARRAY) {
+                return p.list();
+            } else if (p.currentToken() == XContentParser.Token.START_OBJECT) {
+                return p.map();
+            }
+            throw new XContentParseException("Unsupported token [" + p.currentToken() + "]");
```

Comment:
```suggestion
            throw new XContentParseException("Unsupported token [" + p.currentToken() + "]. Expected an array or an object.");
```

nit: We could add a hint, what was actually expected